### PR TITLE
Adds sprite shifting

### PR DESCRIPTION
**Pull request by Shadowtail117**
## About the Pull Request

Original PR is https://github.com/Baystation12/Baystation12/pull/30367. I have received permission from Shadow to use their code.

This PR allows mobs to shift their location on a single tile up to 8 pixels in any direction. Useful mostly for roleplaying, e.g., leaning against a wall or a hug or whatever.

Any form of movement, being buckled, or being grabbed will reset your sprite back to the center.

## Why It's Good For The Game

Allows for more freedom of expression during roleplaying.

## Did you test it?

Yes.

## Changelog

:cl:
rscadd: You can now shift your location on a tile up to 8 pixels in any direction using control + shift + an arrow key.
/:cl:

### CODE REVIEW

File: interface/skin.dmf [hunk 1]
```diff
@@ -1,870 +1,918 @@
...
-	elem 
+	elem
 		name = "ALT+North"
 		command = "northfaceperm"
-	elem 
+	elem
 		name = "CTRL+North"
 		command = "northface"
-	elem 
+		elem
```
Comment (Cupax3) [inline interface/skin.dmf]: Bad indentation I think

**Review by Cupax3 — CHANGES_REQUESTED** [review]
Try fixing that line, maybe the checks will stop screaming after

File: interface/skin.dmf (hunk 1 above)
Comment (Shadowtail117) [reply interface/skin.dmf]: Oops

**Comment by Cupax3** [comment]
Sqosh pls

**Comment by Ilysen** [comment]
I did forget to leave a comment on this, but I do share the same concerns that were brought up in the Bay iteration of this PR. Do you have any ways that you plan to address this, either through code or policy? The `Set-Pose` verb already allows a great deal of freeform expression without any of the potential mechanical problems.

**Comment by Shadowtail117** [comment]
> I did forget to leave a comment on this, but I do share the same concerns that were brought up in the Bay iteration of this PR. Do you have any ways that you plan to address this, either through code or policy?

Apologies, I didn't mention it. The threshold has been halved compared to the original iteration, so you can now be at most a quarter tile away from the center and not a half-tile. Should eliminate ambiguity.

**Comment by Shadowtail117** [comment]
squooomsh

**Review by Cupax3 — APPROVED** [review]
LGTM code-wise, gameplay will have to be simply tested.

**Pull request merged**